src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -166,20 +179,20 @@ public CallSite GetCallSite(int classScope, string methodName, Executable execut
 
             foreach (Guid nodeGuid in nodeGuids)
             {
-                // Get a list of GraphNode objects that correspond to this node.
-                var matchingGraphNodes = graphNodes.Where(gn => gn.guid == nodeGuid);
-
-                if (!matchingGraphNodes.Any())
-                    continue;
-
-                // Get all callsites that match the graph node ids.
-                var matchingCallSites = (from cs in CallsiteCache
-                                         from gn in matchingGraphNodes
-                                         where string.Equals(cs.Key, gn.CallsiteIdentifier)
-                                         select cs.Value);
-
-                // Append each callsite element under node element.
-                nodeMap[nodeGuid] = matchingCallSites.ToList();
```

Comment:
Not sure what the O notation for the existing loop up was but it was bad

Reply:
so looking at this - it appears you are completely skipping the node to graphNode lookup?